File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        int[] src = new int[1800];
        int len = arr.length;
        int count = 0;
        for (int i = 0; i < len; i++)
        {
            if (flag[i])
            {
                for (int j = 0; j < arr[i] * 2; j++)
                    src[count++] = arr[i];
            }
            else
                count -= arr[i];
        }
        int[] answer = new int[count];
        System.arraycopy(src, 0, answer, 0, count);
        return answer;
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {    
    var answer = [];
    for (var i = 0; i < flag.length; i++) {
        if (flag[i])
            answer.push(... new Array(arr[i] * 2).fill(arr[i]));
        else
            answer.splice(-arr[i]);
    }
    return answer;
}